### DISCUSSION TITLE
combo内的节点为何更新不了？

### DISCUSSION THREAD

**Discussion by changxuexu** [antvis/G6#6636] [Q&A]
let curNode = graph.getNodeData(contextmenuEvent.target.id)
  if(curNode['combo']){
    delete curNode['combo']
    graph.updateNodeData([curNode])
    graph.draw();
  }

一个组合内的节点这种写方法为何不能更新？就是想从组合中去除某个节点？

**Comment by yvonneyx** [comment]
是想删除节点还是解除节点与 Combo 的嵌套关系呢

- 删除节点可以直接使用：`graph.removeNodeData()`
- 解除与 combo 的关系：`graph.updateNodeData([{ combo: undefined }])`

**Reply by changxuexu** [reply]
就是解除与 combo 的关系问题。可以参考 [官网案例](https://g6.antv.antgroup.com/zh/examples/element/combo/#rect)
在下面加上下面这段代码：
setTimeout(()=>{
  graph.updateNodeData([
    {
      id: 'node1',
      combo:undefined
    }
  ])
  graph.draw()
  alert(11111)
},5000)

代码执行后，**node1节点依然在combo中**，使用graph.draw()或者graph.render() 都没用
